### DISCUSSION TITLE
Can I automatically assign a label?

### DISCUSSION THREAD

**Discussion by MattDemers** [FreshRSS/FreshRSS#4947] [Q&A] [answered]
I'm migrating over from TinyTinyRSS, and there's the ability to Regex search for feed items, and apply a label to them on intake. 

For instance, with Hacker News' RSS I'd love to be able to take "Show HN" and "Ask HN" topics and automatically apply labels to them, for separation purposes. That way, I can click onto the label and see them.

Does that functionality exist here? I know labels exist, but it seems like adding feed items to them is a manual process.

**Comment by RamenJunkie** [comment]
I would like to second this, even if it's just adding a default label for an entire feed.  I believe the labels appear int he output RSS of favorited posts already.  I run a little script that posts a daily digest to my blog, but having built in labels I could use for tags there or even splitting the feed to several different blogs would be awesome.

**Comment by LivInTheLookingGlass** [comment]
I would also love this. It would essentially replace the need for subcategories

**Comment by Nebucatnetzer** [comment]
I haven’t migrated from TT-RSS yet but would like to. However I heavily rely on applying labels automatically. 
This way I filter out all the news stuff that doesn’t interest me or even filter the content from YouTube channels.

**Reply by Alkarex** [reply]
To filter out irrelevant articles, it is also possible to automatically mark articles as read based on filters, and that can be done at feed level, category level and global level (category and global levels are new since 1.23.0 https://github.com/FreshRSS/FreshRSS/pull/5942 )
Applying labels automatically is also possible now https://github.com/FreshRSS/FreshRSS/pull/5954

**Comment by Nebucatnetzer** [comment]
It‘s probably not possible yet. 
https://github.com/FreshRSS/FreshRSS/issues/2380

**Reply by Alkarex** [reply]
Indeed not possible yet, except with a custom extension, but pretty high on my own todo-list, so will come soon

**Reply by Nebucatnetzer** [reply]
Sounds great, I really like what I‘ve seen so far from FreshRSS.

**Reply by Alkarex** [reply]
Done https://github.com/FreshRSS/FreshRSS/pull/5954

**Comment by Alkarex** [comment, marked as answer]
Ready for testing https://github.com/FreshRSS/FreshRSS/pull/5954
Will soon land in `edge` and will then be in the next release.

**Reply by Nebucatnetzer** [reply]
Great looking forward to testing it when it lands in Nixpkgs.

**Reply by Alkarex** [reply]
If possible, try to test *before* it is released, otherwise it is kind of too late ;-)

**Reply by Nebucatnetzer** [reply]
I know but I’m using the version from the NixOS repository.
I try to update it locally.

**Comment by Alkarex** [comment]
Regex filters https://github.com/FreshRSS/FreshRSS/pull/6706

**Comment by zoechi** [comment]
Is there a way to apply labels to already fetched articles?
It looks like the filter applied added to labels causes labels only to be applied to articles fetched afterwards.

**Reply by Alkarex** [reply]
Not at the moment, but this has been demanded and I guess we already have an issue somewhere. Please link to it if you can find it

**Reply by zoechi** [reply]
This is a recent one I found #6976
